custom/icds_reports/reports/infantometer.py

```diff
@@ -0,0 +1,232 @@
+        'green': OrderedDict()
+    }
+
+    dates = [dt for dt in rrule(MONTHLY, dtstart=three_before, until=month)]
+
+    for date in dates:
+        miliseconds = int(date.strftime("%s")) * 1000
+        data['blue'][miliseconds] = {'y': 0, 'all': 0, 'in_month': 0}
+
+    best_worst = defaultdict(lambda: {
+        'in_month': 0,
+        'all': 0
+    })
+    for row in chart_data:
+        date = row['month']
+        in_month = (row['in_month'] or 0)
+        location = row['%s_name' % loc_level]
+        valid = row['all']
+
+        best_worst[location]['in_month'] = in_month
+        best_worst[location]['all'] = (valid or 0)
+
+        date_in_miliseconds = int(date.strftime("%s")) * 1000
+
+        data['blue'][date_in_miliseconds]['all'] += (valid or 0)
+        data['blue'][date_in_miliseconds]['in_month'] += in_month
+
+    all_locations = [
+        {
+            'loc_name': key,
+            'percent': (value['in_month'] * 100) / float(value['all'] or 1)
+        }
+        for key, value in best_worst.items()
+    ]
+    all_locations_sorted_by_name = sorted(all_locations, key=lambda x: x['loc_name'])
+    all_locations_sorted_by_percent_and_name = sorted(
+        all_locations_sorted_by_name, key=lambda x: x['percent'], reverse=True)
+
+    return {
+        "chart_data": [
+            {
+                "values": [
+                    {
+                        'x': key,
+                        'y': value['in_month'] / float(value['all'] or 1),
+                        'in_month': value['in_month']
+                    } for key, value in data['blue'].items()
+                ],
+                "key": "Percentage of AWCs that reported having an Infantometer",
+                "strokeWidth": 2,
+                "classed": "dashed",
+                "color": ChartColors.BLUE
+            },
+        ],
+        "all_locations": all_locations_sorted_by_percent_and_name,
+        "top_five": all_locations_sorted_by_percent_and_name[:5],
+        "bottom_five": all_locations_sorted_by_percent_and_name[-5:],
+        "location_type": loc_level.title() if loc_level != LocationTypes.SUPERVISOR else 'Sector'
+    }
+
+
+@icds_quickcache(['domain', 'config', 'loc_level', 'location_id', 'show_test'], timeout=30 * 60)
+def get_infantometer_sector_data(domain, config, loc_level, location_id, show_test=False):
+    group_by = ['%s_name' % loc_level]
+
+    config['month'] = datetime(*config['month'])
+    data = AggAwcMonthly.objects.filter(
+        **config
+    ).values(
+        *group_by
+    ).annotate(
+        in_month=Sum('infantometer'),
+        all=Sum('num_awc_infra_last_update')
+    ).order_by('%s_name' % loc_level)
+
+    if not show_test:
+        data = apply_exclude(domain, data)
+
+    chart_data = {
+        'blue': [],
+    }
+
+    tooltips_data = defaultdict(lambda: {
+        'in_month': 0,
+        'all': 0
+    })
+
+    loc_children = get_child_locations(domain, location_id, show_test)
```

Comment:
Can you remove  this set of lines. These are causing problems on UI and now adding these does not make any sense as we have started using SQLLocation class as the source of location for aggregation.
Example removal is here: https://github.com/dimagi/commcare-hq/pull/25692/files#diff-87fe07900cef15630f5108f0346a2ddeL120
@challabeehyv